src/client.ts

```diff
@@ -458,6 +459,70 @@ export class GenesysCloudWebrtcSdk extends (EventEmitter as { new(): StrictEvent
     }
   }
 
+    /**
+   * Update the default resolution of the selected video track for the sdk.
+   *  Pass in the following:
+   *  - resolution: Either undefined or an object containing ConstrainULongs
+   *      representing both width and height
+   *  - updateActiveSessions: boolean
+   *
+   * If `updateActiveSessions` is `true`, any active sessions will
+   *  have their video resolutions updated to either the selected resolution
+   *  or the next best resolution the device is capable of
+   *
+   * Else, only the sdk defaults will be updated and active sessions'
+   * resolutions will not be touched
+   *
+   * Success or failure of `applyConstraints` will emit an event for
+   * the consuming app to listen for so that it can determine if the
+   * resolution they requested is actually in place or the next best
+   * option. This will allow the consuming app to notify the user as
+   * they see fit.
+   *
+   * @param resolution the width and height that was requested
+   * to update the video; could be undefined to represent the default value
+   *
+   * @returns a promise that fullfils once the event has been emitted
+   * signaling that the resolution has updated
+   */
+  async updateDefaultResolution(resolution: IVideoResolution | undefined, updateActiveSessions: boolean): Promise<any> {
+    this._config.defaults.videoResolution = resolution;
+    if (updateActiveSessions) {
+      this.sessionManager.getAllActiveSessions().filter(session =>
+        session.sessionType === 'collaborateVideo').forEach(videoSession => {
+          videoSession._outboundStream.getVideoTracks().forEach(async track => {
```

Comment:
Coupld things: 
1. NIT: I think the indentation could be a little different to help keep the functions and their callbacks a little more clear.
2. I am not 100% sure, but I _think_ there are instances when we could have a session but no _outboundStream (maybe for 1-1 prejoin). It may be good to put in a null check into the filter. 

``` ts
   this.sessionManager.getAllActiveSessions()
      .filter(session => session.sessionType === 'collaborateVideo' && session._outboundStream)
      .forEach(videoSession => {
        videoSession._outboundStream.getVideoTracks().forEach(async track => { 
        // ...
```

Reply:
> 2. I am not 100% sure, but I _think_ there are instances when we could have a session but no _outboundStream (maybe for 1-1 prejoin). It may be good to put in a null check into the filter.

You're right.